Dockerfile

```diff
@@ -1,19 +1,39 @@
+# build stage
+FROM python:3.9 AS builder
+
+RUN mkdir /app \
+ && chmod 700 /app
+
+COPY poetry.lock /app/
+COPY pyproject.toml /app/
+
+WORKDIR /app
+
+RUN python3.9 -m pip install poetry \
+ && python3.9 -m poetry config virtualenvs.create false \
+ && python3.9 -m poetry install --without dev \
+ && python3.9 -m poetry export -f requirements.txt --output requirements.txt --without-hashes
+
+# run stage
 FROM quay.io/centos/centos:stream8
 
 RUN dnf -y module install python39 && dnf -y install python39 python39-pip
-RUN mkdir /app
-ADD https://raw.githubusercontent.com/arcalot/arcaflow-plugin-template-python/main/LICENSE /app
-ADD example_plugin.py /app
-ADD test_example_plugin.py /app
-ADD requirements.txt /app
+RUN mkdir /app \
+ && chmod 700 /app
+
+COPY --from=builder /app/requirements.txt /app/
+COPY LICENSE /app/
+COPY README.md /app/
+COPY example_plugin.py /app/
+COPY test_example_plugin.py /app/
```

Comment:
The way this is doing it is building the requirements and using them in the second step. But yeah, it may be a better idea to build a venv and just use that in the other stage.

Reply:
I have attempted to build and pass the venv between stages with the recent push